### PR TITLE
add some opencl related libraries

### PR DESCRIPTION
**Pull request by glehmann**

### CODE REVIEW

File: Library/Formula/clblas.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  depends_on "boost" => :build
+
+  patch do
+    # install clBLAS cmake configuration and version files
+    url "https://github.com/clMathLibraries/clBLAS/pull/68.diff"
+    sha1 "b85a17cd972ed598a80aef8623830c90f5abb1f4"
+  end
+
+  def install
+    system "cmake", "-DBUILD_TEST=OFF", "-DBUILD_CLIENT=ON", "-DBUILD_KTEST=OFF", "-DSUFFIX_LIB:STRING=", "-DCMAKE_MACOSX_RPATH:BOOL=ON", "src", *std_cmake_args
```
Comment (DomT4) [inline Library/Formula/clblas.rb]: Might be worth separating this into an array to make it more readable.

```ruby

args = std_cmake_args
args << "-DBUILD_TEST=OFF" << "-DBUILD_CLIENT=ON" << "-DBUILD_KTEST=OFF"
args << "-DSUFFIX_LIB:STRING=" << "-DCMAKE_MACOSX_RPATH:BOOL=ON" << "src"

system "cmake", *args
```

Or similar.

Comment (glehmann) [inline Library/Formula/clblas.rb]: thanks for the suggestion :)

File: Library/Formula/arrayfire.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
+class Arrayfire < Formula
+  homepage "http://arrayfire.com"
+  url "https://github.com/arrayfire/arrayfire/archive/v3.0beta.tar.gz"
+  version "3.0beta"
```
Comment (mikemcquaid) [inline Library/Formula/arrayfire.rb]: This should be a stable version for us to include it.

File: Library/Formula/arrayfire.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  depends_on "pkg-config" => :build
+  depends_on "freeimage"
+  depends_on "fftw"
+  depends_on "clblas"
+  depends_on "clfft"
+
+  patch do
+    url "https://github.com/arrayfire/arrayfire/pull/454.diff"
+    sha256 "cf862821a88ec979d146a9acf34728ef294ec658ed29c1d0adb7af4a5c7bed69"
+  end
```
Comment (mikemcquaid) [inline Library/Formula/arrayfire.rb]: Can you please add an explanation of why the patch is needed in a comment in the formula file. Thanks!

File: Library/Formula/arrayfire.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        int device = argc > 1 ? atoi(argv[1]) : 0;
+        af::setDevice(device);
+        af::info();
+        af::array A = af::randu(5,3, f32);
+        af_print(A);
+        return 0;
+      }
+    EOS
+    system ENV.cxx, "-o", "test_cpu", "-I#{include}", "-L#{lib}", "-lafcpu", testpath/"test.cpp"
+    system "./test_cpu"
```
Comment (mikemcquaid) [inline Library/Formula/arrayfire.rb]: :+1: nice test

File: Library/Formula/boost-compute.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      {
+        std::cout << "hello from "
+          << boost::compute::system::default_device().name() << std::endl;
+        return 0;
+      }
+    EOS
+    system ENV.cxx, "-o", "hello", "-I#{include}/compute", "-framework", "OpenCL", testpath/"hello.cpp"
+    output = `./hello`
+    assert $?.success?
+    print output
```
Comment (mikemcquaid) [inline Library/Formula/boost-compute.rb]: Probably want to kill the `print` here?

File: Library/Formula/boost-compute.rb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      #include <boost/compute/core.hpp>
+      int main()
+      {
+        std::cout << "hello from "
+          << boost::compute::system::default_device().name() << std::endl;
+        return 0;
+      }
+    EOS
+    system ENV.cxx, "-o", "hello", "-I#{include}/compute", "-framework", "OpenCL", testpath/"hello.cpp"
+    output = `./hello`
```
Comment (mikemcquaid) [inline Library/Formula/boost-compute.rb]: Use `shell_output` which asserts on the success and then use `assert_match` below.

File: Library/Formula/clblas.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  homepage "https://github.com/clMathLibraries/clBLAS"
+  url "https://github.com/clMathLibraries/clBLAS/archive/v2.2.tar.gz"
+  sha256 "0563e028485d240f8b6e6efcae61677049627b19e924dcb1ef8014065b6416df"
+
+  depends_on "cmake" => :build
+  depends_on "boost" => :build
+
+  patch do
+    # install clBLAS cmake configuration and version files
+    url "https://github.com/clMathLibraries/clBLAS/pull/68.diff"
```
Comment (mikemcquaid) [inline Library/Formula/clblas.rb]: Probably want this to get accepted upstream before we accept it.

File: Library/Formula/clblas.rb [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    sha1 "b85a17cd972ed598a80aef8623830c90f5abb1f4"
+  end
+
+  def install
+    opts = std_cmake_args
```
Comment (mikemcquaid) [inline Library/Formula/clblas.rb]: `args` is generally preferred.

File: Library/Formula/clfft.rb [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    # don't use lib64 in lib path
+    url "https://github.com/clMathLibraries/clFFT/pull/61.diff"
+    sha1 "b4cec7072c592a68c5c603d6f2e628baaedca35e"
+  end
+
+  patch do
+    # install cmake config and version files
+    url "https://github.com/clMathLibraries/clFFT/pull/68.diff"
+    sha256 "31e5a6ffc8e94e30970035162c2b45418795929402513b189d2f8fd3a3810ef0"
+  end
```
Comment (mikemcquaid) [inline Library/Formula/clfft.rb]: Again, probably want to wait for all these PRs to land before including.

File: Library/Formula/arrayfire.rb (hunk 2 above)
Comment (glehmann) [inline Library/Formula/arrayfire.rb]: ok, I'll wait for the stable release :-)

File: Library/Formula/boost-compute.rb (hunk 5 above)
Comment (glehmann) [inline Library/Formula/boost-compute.rb]: :+1:

File: Library/Formula/clblas.rb (hunk 7 above)
Comment (glehmann) [inline Library/Formula/clblas.rb]: It's already accepted. I'll replace it by the actual commit.

File: Library/Formula/clfft.rb (hunk 9 above)
Comment (glehmann) [inline Library/Formula/clfft.rb]: They have all been accepted too. I prepared the formulas before they were accepted. I'll replace them with the actual commits.

**Comment by glehmann** [comment]
Tests are all green :-)

Could you review and (hopefully) merge the PR?

**Comment by jacknagel** [comment]
some/all of these seem more appropriate for homebrew-science?

**Comment by glehmann** [comment]
clfft and clblas would definitely fit better in homebrew-science. boost-compute and clinfo are more generic tools though. Should I move them also to homebrew-science?

**Comment by glehmann** [comment]
clfft and clblas moved to https://github.com/Homebrew/homebrew-science/pull/2192
Just let me know if this is ok to merge these two remaining ones or not :-)

**Comment by glehmann** [comment]
moving to https://github.com/Homebrew/homebrew-science/pull/2353

**Pull request closed without merging**